AST.fs

```diff
@@ -500,7 +513,7 @@ let (|BoolExp'|ArithExp'|AnyExp'|) (e : Expression')
     | ArraySubscript _ -> AnyExp' e
     | Num _ -> ArithExp' e
     | True | False -> BoolExp' e
-    | BopExpr(BoolOp, _, _) -> BoolExp' e
+    | BopExpr(BoolOp, _, _) | UopExpr(_,_) -> BoolExp' e
```

Comment:
This might need refining if we ever put in any non-Boolean unary operators (pointers?)